app/lib/actions/pulp3/repository/delete.rb

```diff
@@ -0,0 +1,16 @@
+module Actions
+  module Pulp3
+    module Repository
+      class Delete < Pulp3::Abstract
+        def plan(repository, smart_proxy)
+          plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id)
+        end
+
+        def run
+          repo = ::Katello::Repository.find(input[:repository_id])
+          output[:response] = repo.backend_service(smart_proxy).delete
+        end
```

Comment:
I think this task is async, so you'd want to use  Pulp3::AbstractAsyncTask similar to RefreshDistribution